src/unity/toolkits/supervised_learning/automatic_model_creation.cpp

```diff
@@ -1,44 +1,128 @@
+#include <unity/lib/api/unity_sframe_interface.hpp>
 #include <unity/toolkits/supervised_learning/automatic_model_creation.hpp>
-#include <unity/toolkits/supervised_learning/supervised_learning.hpp>
 #include <unity/toolkits/supervised_learning/boosted_trees.hpp>
+#include <unity/toolkits/supervised_learning/decision_tree.hpp>
+#include <unity/toolkits/supervised_learning/linear_regression.hpp>
+#include <unity/toolkits/supervised_learning/linear_svm.hpp>
+#include <unity/toolkits/supervised_learning/logistic_regression.hpp>
+#include <unity/toolkits/supervised_learning/random_forest.hpp>
+#include <unity/toolkits/supervised_learning/supervised_learning.hpp>
 
 
 namespace turi {
 namespace supervised {
 
-using namespace xgboost; 
+std::shared_ptr<supervised_learning_model_base> create_classifier(const std::string& model_name) {
+  std::shared_ptr<supervised_learning_model_base> result;
+  if(model_name == "boosted_trees_classifier") {
+    result = std::make_shared<xgboost::boosted_trees_classifier>();
+  } else if (model_name == "random_forest_classifier") {
+    result = std::make_shared<xgboost::random_forest_classifier>();
+  } else if (model_name == "decision_tree_classifier") {
+    result = std::make_shared<xgboost::decision_tree_classifier>();
+  } else if (model_name == "classifier_logistic_regression") {
+    result = std::make_shared<logistic_regression>();
+  } else if (model_name == "classifier_svm") {
+    result = std::make_shared<linear_svm>();
+  } else {
+    std::ostringstream error_msg;
+    error_msg << "Unrecognized Model: " << model_name;
+    log_and_throw(error_msg.str().c_str());
+  }
+  return result;
+}
+
+
+// Return the validation accuracy if it exists, otherwise return training accuracy.
+double get_classifier_accuracy(std::shared_ptr<supervised_learning_model_base> model) {
+  flexible_type accuracy = FLEX_UNDEFINED;
+
+  // Get the model's progress SFrame.
+  variant_type progress = model->get_value_from_state("progress");
+  assert(get_variant_which_name(progress.which()) == "SFrame");
```

Comment:
Use our ASSERT_TRUE statement instead of the C assert call, which just aborts the program.

Reply:
Done. Changed two other `assert`s at the end of this function too.